### DISCUSSION TITLE
DOMPDF (2.0.0) PNG Image loading issue . PHP version 8.0.17

### DISCUSSION THREAD

**Discussion by ayushmohanan** [dompdf/dompdf#2956] [General]
Hey all, 

I am having an issue with DOMPDF image loading. I tried all the way to load the image but it won't work on my local / server. 
When I try with the full URL it is returning an "Allowed memory size" error.
isRemoteEnabled mode changed to true . 
I wanted to load PNG image. 
Please help me if you have an idea of how to resolve this issue. 
I really appreciate any help you can provide. 

My Code Sample
constructor 
```
    $options = new Options();
        $options->set('defaultFont', 'Helvetica');
        $options->set('isRemoteEnabled', TRUE);
        $options->set('isHtml5ParserEnabled', true);
        $this->dompdf = new Dompdf($options);
```

Contoller
```
        $this->response->removeHeader('Content-Type');
        #$this->dompdf->set_base_path($bp); 
        #return view($this->path_views . 'contract', $data);
        #viewpage data
        $this->dompdf->loadHtml(view($this->path_views . 'contract', $data));
        $this->dompdf->set_option('isRemoteEnabled', TRUE);
        $this->dompdf->set_option('defaultMediaType', 'all');
        $this->dompdf->set_option('isFontSubsettingEnabled', true);
        $this->dompdf->set_option('isPhpEnabled', true);
        $this->dompdf->setPaper('A4','portrait');
        $this->dompdf->render();
        #$pdf = $this->dompdf->output();
        #file_put_contents("contract", $pdf);
        $this->dompdf->stream( $contract, array("Attachment" => false));
```
 view page 
```
     <td width="20%"> <!--<img src="../../../logo.png" /> --></td>
```

**Comment by acgtwentyone** [comment]
it would help if you provide more details about the error you are getting. Such as the print of it. 
Is "Allowed memory size" error the full error you are getting ?

**Reply by ayushmohanan** [reply]
Hi @acgtwentyone,

 Thanks for your reply, 

Yes, this is the error am getting when I try to print the base64 image on the view. 

![ErrorException](https://user-images.githubusercontent.com/44460113/179515130-7d9a67bc-8b7e-4a96-8cb4-1fc252564552.png)

**Comment by bsweeney** [comment]
How large is your image? And what is the PHP memory limit?

**Reply by ayushmohanan** [reply]
Hi @bsweeney , 

My image size is very low  72 KB only. And the PHP memory limit is 512M .  For me without an image pdf is loading properly. 
![Fireman-PDF-Manage-pdf (2)](https://user-images.githubusercontent.com/44460113/179515625-37b8e6d4-6c8e-43b3-8779-0050a893d58b.png)

**Reply by bsweeney** [reply]
What type of PNG is this? Can you post it here so I can test with it?

**Reply by ayushmohanan** [reply]
Sure , 

 This is the png image I have used. 

![logo](https://user-images.githubusercontent.com/44460113/179661554-186d33a1-3e53-4df2-a8ff-caf2ecfed51d.png)

**Reply by bsweeney** [reply]
Right now I'm at a loss. I really don't see anything about the image that would be problematic. Rendering a document with that image on my own system doesn't significantly increase memory usage.

The image isn't size-constrained in your HTML, meaning Dompdf will try to render it based on its dimensions (initially, at least). If you tweak the image element as follows does it render without running into an error?

```
<img src="<?= $base64; ?>" style="max-width: 100%"/>
```

You can also try to work around the issue by using a version without transparency. Or using a JPG instead, which requires no processing in this use case.

**Comment by ayushmohanan** [comment]
I have updated my code like the following , Still am getting the same error. 
Constructor:

```
  $options = new Options();
  $options->set('defaultFont', 'Helvetica');
  $options->set('isRemoteEnabled', TRUE);
  $options->set('debugKeepTemp', true);
  $options->set('isHtml5ParserEnabled', true);
  $this->dompdf = new Dompdf($options);
```

Fucntion :

```
    $path = '../logo.png'; 
    $type = pathinfo($path, PATHINFO_EXTENSION); 
    $data = file_get_contents($path); 
    $base64 = 'data:image/' . $type . ';base64,' . base64_encode($data);

    #data
    $data = array_merge($this->data, [
        'title' => $this->title,
        'word' => $this->word,
        'breadcrumb' => array("0" => array("url" => base_url("home"), "title" => 'Home'), "1" => array("url" => $url,
            "title" => $this->word_multi)),
        'action' => $url,
        'customerID' => $cid,
        'siteID' => $site,
        'contractRow' => $contractRow,
        'base64' => $base64

    ]);     

    #Contract file name  
    $contract = str_replace('/', "-", $contractRow['contractNumber']);        
    #remove UTF8 issue header     
    $this->response->removeHeader('Content-Type');     
    #viewpage data
    $this->dompdf->loadHtml(view($this->path_views . 'contract', $data));
    $this->dompdf->set_option('defaultMediaType', 'all');    
    $this->dompdf->setPaper('A4','portrait');
    $this->dompdf->render(); 
   
    $this->dompdf->stream( $contract, array("Attachment" => false));
```

View
![Capture](https://user-images.githubusercontent.com/44460113/179517291-6975bc27-780f-4a7b-a5a2-7f01e0392d33.PNG)

**Reply by acgtwentyone** [reply]
In this scenario, i think the best way to test it is to eliminate as much information as you could send to view in in order to display only image in the view. Remove or comment a few keys on the $data array for example. Also some configurations should be omitted, such as isRemoteEnabled (if the image is locally). After this procedure if the image is displayed correct maybe the issue is not related to the image itself, maybe is something else causing the issue.
